crates/release_plz_core/src/release.rs

```diff
@@ -131,9 +140,41 @@ fn release_package(
         info!("published {}", package.name);
     }
 
+    if input.gh_release {
+        publish_release(git_tag(&package.name, &package.version.to_string()), input);
+    }
+
     Ok(())
 }
 
+async fn publish_release(git_tag: String, input: &ReleaseRequest) {
+    let repo_url = match &input.repo_url {
+        Some(url) => RepoUrl::new(url.as_str()),
+        None => {
+            let repo = Repo::new(input.workspace_root().unwrap()).unwrap();
+            let url = repo
+                .origin_url()
+                .context("cannot determine origin url")
+                .unwrap();
+            RepoUrl::new(&url)
+        }
+    }
+    .unwrap();
+    let secret_string = SecretString::from(input.git_token.to_string());
+    let github = GitHub::new(repo_url.clone().owner, repo_url.clone().name, secret_string);
+    let github_client = GitHubClient::new(&github);
+    if repo_url.is_on_github() {
+        let page = Octocrab::default()
+            .repos(repo_url.clone().owner, repo_url.name)
+            .releases()
+            .create(&git_tag)
+            .send()
+            .await
+            .unwrap();
```

Comment:
```suggestion
    let github_client = GitHubClient::new(&github).unwrap();
    if repo_url.is_on_github() {
        let page = github_client.create_release(&git_tag).await.unwrap();
```
Now that we have the github_client, we can use it to create the release!

I have extracted the release functionality into the `create_release` function. You can add this function [here](https://github.com/MarcoIeni/release-plz/blob/f03c598c38e8cf5ceeae542add5d54fae3915159/crates/release_plz_core/src/github_client.rs#L109):
```rust
    pub async fn create_release(&self, tag: &str) -> anyhow::Result<repos::Release> {
        self.client
            .repos(&self.github.owner, &self.github.repo)
            .releases()
            .create(tag)
            .send()
            .await
            .context("Failed to create release")
    }
```